skbio/tree/tests/test_tree.py

```diff
@@ -598,8 +600,22 @@ def test_tip_tip_distances_non_tip_endpoints(self):
 
     def test_tip_tip_distances_no_length(self):
         t = TreeNode.read(StringIO(u"((a,b)c,(d,e)f);"))
-        with self.assertRaises(NoLengthError):
-            t.tip_tip_distances()
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter("error")
+            with self.assertRaises(RepresentationWarning):
+                t.tip_tip_distances()
```

Comment:
@MikeDeberg tried this at first, but execution of the block ends as soon as the `RepresentationWarning` is sent, so variables never get set, etc... You'll notice that this is checked in the test below because we just ignore the warning. (Do you know a way around this?)